scio-test/src/main/scala/com/spotify/scio/testing/SCollectionMatchers.scala

```diff
@@ -35,16 +38,42 @@ import scala.reflect.ClassTag
  */
 trait SCollectionMatchers {
 
+  import scala.language.higherKinds
+
+  sealed trait MatcherBuilder[T, B, A[_]] {
+    self: Matcher[T] =>
+    type AssertBuilder = A[B] => A[B]
+
+    def matcher(builder: AssertBuilder): Matcher[T]
+
+    def matcher: Matcher[T] = matcher(identity)
```

Comment:
Where is this used? Don't see any parameter-less `.matcher`.

Reply:
@nevillelyh they are used in `IterableMatcher` and `SingleMatcher`

Reply:
LOL yeah it expands to `.matcher(identity).apply(left)`.